chain/rosetta-rpc/src/adapters/transactions.rs

```diff
@@ -154,35 +154,52 @@ pub(crate) fn convert_block_changes_to_transactions(
     exec_to_rx: ExecutionToReceipts,
 ) -> crate::errors::Result<RosettaTransactionsMap> {
     let mut transactions = RosettaTransactions::new(exec_to_rx, block_hash);
-    for account_change in accounts_changes {
-        let transaction = transactions.get_for_cause(account_change.cause)?;
-        let operations = &mut transaction.operations;
-        match account_change.value {
+
+    // Sort changes by account so that we have deterministic ordering of
+    // operations.
+    let mut changes_by_account = std::collections::BTreeMap::new();
+    for change in accounts_changes {
+        let (account_id, account) = match change.value {
             near_primitives::views::StateChangeValueView::AccountUpdate { account_id, account } => {
-                let previous_account_state = accounts_previous_state.get(&account_id);
+                (account_id, Some(account))
+            }
+            near_primitives::views::StateChangeValueView::AccountDeletion { account_id } => {
+                (account_id, None)
+            }
+            unexpected_value => {
+                return Err(crate::errors::ErrorKind::InternalInvariantError(format!(
+                    "queried AccountChanges, but received {:?}.",
+                    unexpected_value
+                )))
+            }
+        };
+        changes_by_account.entry(account_id).or_insert_with(Vec::new).push((change.cause, account));
+    }
+
+    for (account_id, changes) in changes_by_account {
+        let mut previous_account_state = accounts_previous_state.remove(&account_id);
```

Comment:
there is a behavior change:

in previous code - accounts_previous_state had 'correct' values after the end of the run (as we were doing 'inserts').

Now - this accounts_previous_state will be missing values for every entry that we touched.

if this is working as intended - we should mention it in the function description.

Reply:
`accounts_previous_state` is moved by value into the function and never returned. It doesn’t matter that it’s value changes and it’s not a behaviour that is observable by callers of the function.